security/keycloak-oidc-client-extended/src/main/java/io/quarkus/ts/security/keycloak/oidcclient/extended/restclient/ping/filters/DefaultTokenRequestFilter.java

```diff
@@ -0,0 +1,13 @@
+package io.quarkus.ts.security.keycloak.oidcclient.extended.restclient.ping.filters;
+
+import io.quarkus.oidc.token.propagation.AccessTokenRequestFilter;
+
+/**
+ * This class is required for
+ * {@link io.quarkus.ts.security.keycloak.oidcclient.extended.restclient.ping.clients.TokenPropagationPongClient}
+ * It would not be required normally, but having {@link CustomTokenRequestFilter} causes AmbiguousResolutionException when
+ * getting a default filter.
+ * So this class is necessary to have unambiguous filter for TokenPropagatingPongClient.
```

Comment:
Sounds like something that could be fixed with qualifiers since we know this at build time. Can you please open issue? IMO this is not expected behavior, because Quarkus knows there is need to use `AccessTokenRequestFilter`. The worst case scenario, Quarkus can use annotation transformer and produce this class for you.

Reply:
Please ping me there, I'd like to have a look at it.

Reply:
I agree this behaviour is not OK. I've created an issue https://github.com/quarkusio/quarkus/issues/36994

Reply:
Thank you